ow_power_system/src/power_system_node.cpp

```diff
@@ -17,27 +25,117 @@ int main(int argc, char* argv[]) {
 
   //Construct our power draw listener
   //TODO: Check out latching
-  ros::Subscriber power_listener = n.subscribe("power_draw", 1000, computeSOC);
+  //ros::Subscriber power_listener = n.subscribe("power_draw", 1000, computeSOC);
 
-  //Construct our State of Charge (SOC) publisher
+  //Construct our State of Charge (SOC) publisher (Volts)
   ros::Publisher SOC_pub = n.advertise<std_msgs::Float64>("state_of_charge",1000);
+  //Construct our Remaining Useful Life (RUL) publisher (Seconds)
+  ros::Publisher RUL_pub = n.advertise<std_msgs::Float64>("remaining_useful_life",1000);
+  
+  //Load power values csv
+  string csv_path,csv_file;
+  csv_file = "/data/onewatt.csv";
+  //FILE OPTIONS: onewatt.csv, eightwatt.csv, sixteenwatt.csv
+  csv_path = ros::package::getPath("ow_power_system");
+  csv_path += csv_file;
+
+  auto power_csv = loadCSV(csv_path);
+  
+  //Define our publication rate and initialize our time to 0
+  double power_update_rate;  
+  if (csv_file=="/data/onewatt.csv"){
+    power_update_rate = 0.10; //csv is at 10Hz intervals
+  } else {
+    power_update_rate = 1.0; // 1 Hz default
+  }
+  int t_step= 1;//starting at 1 not zero to skip csv headings
```

Comment:
based on loadCSV code, I think this t_step should be set to zero

```suggestion
  int t_step= 0;
```